### PR TITLE
Fixes auth failure in omiserver

### PR DESCRIPTION
**Pull request by yakman2020**
We need to include an anonymous credential in gss_accept_sec_context as a verifier cred.

### CODE REVIEW

**Comment by yakman2020** [comment]
@Microsoft/omi-devs 
@jeffaco

File: Unix/http/httpauth.c [hunk 1]
```diff
@@ -1529,9 +1538,12 @@ MI_Boolean IsClientAuthorized(_In_ Http_SR_SocketData * handler)
 
                 (* _g_gssState.Gss_Delete_Sec_Context)(&min_stat, &context_hdl, NULL);
 
-                handler->pAuthContext = NULL;
+                handler->pAuthContext  = NULL;
```
Comment (jeffaco) [inline Unix/http/httpauth.c]: Why add a second space here, when it's inconsistent from all else here?

File: Unix/http/httpclientauth.c [hunk 2]
```diff
@@ -684,11 +684,12 @@ static char *_BuildInitialGssAuthHeader(_In_ HttpClient_SR_SocketData * self, MI
     // gss_OID_set_desc mechset_krb5 = { 1, &mech_krb5 };
     // gss_OID_set_desc mechset_iakerb = { 1, &mech_iakerb };
     const gss_OID_desc mechset_avail_elems[] = {
+        { 6, "\053\006\001\005\005\002" },
         { 10, "\x2b\x06\x01\x04\x01\x82\x37\x02\x02\x0a" }, // For now we start with ntlm
         // mech_krb5,   Not yet
         // mech_iakerb,  Not yet
     };
-    const gss_OID_set_desc mechset_avail = { /*3, */ 1, (gss_OID) mechset_avail_elems };
+    const gss_OID_set_desc mechset_avail = { 2, /* 1 ,*/ (gss_OID) mechset_avail_elems };
```
Comment (jeffaco) [inline Unix/http/httpclientauth.c]: Why the `/* `, /*` part? Not certain of this? If certain, remove it?

File: Unix/http/httpauth.c [hunk 3]
```diff
@@ -1529,9 +1538,12 @@ MI_Boolean IsClientAuthorized(_In_ Http_SR_SocketData * handler)
...
                 handler->authFailed = TRUE;
 
+                (* _g_gssState.Gss_Release_Cred)(&min_stat, handler->pVerifierCred);
```
Comment (jeffaco) [inline Unix/http/httpauth.c]: No check if `handler->pVerifierCred` is null or not. How does Gss_Release_cred behave in this case? I also noted that `gss_release_cred` is called without checking the return value. This is done in other cases too, for other gss APIs. Is this a good idea? Even if you just note the error and log it and then continue, having this in the log can provide critical information if things go awry.

File: Unix/http/httpclientauth.c (hunk 2 above)
Comment (jeffaco) [inline Unix/http/httpclientauth.c]: This is littered with "magic numbers", like at line 687, which you added. Comments as to what these are would be appreciated!

File: Unix/http/httpauth.c (hunk 3 above)
Comment (yakman2020) [inline Unix/http/httpauth.c]: I'm checking it as good manners, but there is a trace message just above it.  pVerifierCred essentially couldn't be null here.

File: Unix/http/httpclientauth.c (hunk 2 above)
Comment (yakman2020) [inline Unix/http/httpclientauth.c]: good point. Added comments.

**Pull request merged**